pkgs/development/python-modules/numpy/default.nix

```diff
@@ -28,6 +28,9 @@ in buildPythonPackage rec {
   nativeBuildInputs = [ gfortran pytest ];
   buildInputs = [ blas ];
 
+  # If the blas imlementation is MKL, we need to propagate it for the setup-hook.
+  propagatedBuildInputs = [ ] ++ lib.optional (blasImplementation == "mkl") blas;
```

Comment:
Since callers all the way up to the top of the stack will need to run `mkl`'s `setup-hook`, it has to be propagated as far as I can tell.